src/analysis/retail/warlock/demonology/modules/talents/DreadCalling.tsx

```diff
@@ -0,0 +1,60 @@
+import { formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_WARLOCK } from 'common/TALENTS';
+import { TooltipElement } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+/**
+ * Each Soul Shard spent on Hand of Gul'dan increases the damage of your next Call Dreadstalkers by 4%.
+ */
+
+class DreadCalling extends Analyzer {
+  callDreadstalkersCasts: number = 0;
+  dreadCallingStacksConsumed: number = 0;
+
+  DAMAGE_BUFF_PER_STACK = 0.04;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_WARLOCK.DREAD_CALLING_TALENT);
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CALL_DREADSTALKERS),
+      this.onCallDreadstalkersCast,
+    );
+  }
+
+  onCallDreadstalkersCast(event: CastEvent) {
+    this.callDreadstalkersCasts += 1;
+
+    this.dreadCallingStacksConsumed += this.selectedCombatant.getBuffStacks(
+      SPELLS.DREAD_CALLING_BUFF.id,
+      event.timestamp,
+    );
+  }
+
+  statistic() {
+    const averageNumberOfStacks = this.dreadCallingStacksConsumed / this.callDreadstalkersCasts;
+    const averageDamageBuff = this.DAMAGE_BUFF_PER_STACK * averageNumberOfStacks;
```

Comment:
Generally we like to use getters for defining these variables so that other files can easily use them in the future if needed, see https://github.com/WoWAnalyzer/WoWAnalyzer/blob/dragonflight/src/analysis/retail/evoker/preservation/modules/talents/Reversion.tsx if you need an example